### PR TITLE
enhancement(ci): reduce billable time of Test Suite

### PR DESCRIPTION
**Pull request by neuronull**
- save time by running unit tests and vrl test not in separate jobs

### CODE REVIEW

**Comment by neuronull** [comment]
tested here: https://github.com/vectordotdev/vector-ci-draft/actions/runs/5327501342/jobs/9650977004

**Comment by neuronull** [comment]
This results in a roughly 20-25% billable runtime reduction for the test suite which runs on each PR push.

**Comment by neuronull** [comment]
>  This results in a roughly 20-25% billable runtime reduction for the test suite which runs on each PR push.

50% now.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -33,50 +33,6 @@ jobs:
...
-    if: ${{ needs.changes.outputs.source == 'true' || needs.changes.outputs.cue == 'true' }}
-    steps:
-      - uses: actions/checkout@v3
-      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
-      - run: bash scripts/environment/prepare.sh
-      - run: cargo vdev test-vrl
-
   checks:
     name: Checks
     runs-on: [linux, ubuntu-20.04-8core]
```
Comment (dsmith3197) [inline .github/workflows/test.yml:38]: Not specifically related to the changes here, but we also want to move up the cache step in this workflow to run before we run `prepare.sh`. As it is currently set up, the cache is being populated after we download and install the cached dependencies.
```
      - uses: actions/cache@v3
        name: Cache Cargo registry + index
        with:
          path: |
            ~/.cargo/bin/
            ~/.cargo/registry/index/
            ~/.cargo/registry/cache/
            ~/.cargo/git/db/
          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
          restore-keys: |
            ${{ runner.os }}-cargo-
      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
      - run: bash scripts/environment/prepare.sh
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -133,6 +100,8 @@ jobs:
       - name: Check Rust Docs
         if: needs.changes.outputs.source == 'true'
         run: cd rust-doc && make docs
+      - if: needs.changes.outputs.source == 'true' || needs.changes.outputs.cue == 'true'
+        run: cargo vdev test-vrl
```
Comment (dsmith3197) [inline .github/workflows/test.yml:104]: We should add a name for this step 🙂 
```suggestion
      - name:  VRL - Linux
        if: needs.changes.outputs.source == 'true' || needs.changes.outputs.cue == 'true'
        run: cargo vdev test-vrl
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (neuronull) [reply .github/workflows/test.yml:38]: good catch 👍

File: .github/workflows/test.yml (hunk 2 above)
Comment (neuronull) [reply .github/workflows/test.yml]: I also added some whitespace to increase readability to better catch that kind of thing.

**Review by jszwedko — APPROVED** [review]
I'm sad to lose the separate jobs, but it seems worth it to cut down on costs. Hopefully we can bring them back when we solve the build caching issue.

**Comment by neuronull** [comment]
>  when

>  we solve the build caching issue.

❤️ when, (not if) ! 🎉

**Review by dsmith3197 — APPROVED** [review]

**Pull request merged**